src/backend/distributed/executor/subplan_execution.c

```diff
@@ -36,7 +36,10 @@ ExecuteSubPlans(DistributedPlan *distributedPlan)
 	List *subPlanList = distributedPlan->subPlanList;
 	ListCell *subPlanCell = NULL;
 	List *nodeList = NIL;
-	bool writeLocalFile = false;
+	ListCell *nodeCell = NULL;
+	bool writeLocalFile = true;
```

Comment:
Might be nice to only set this to true when we actually need it, e.g. for a HAVING clause that uses it.